### ISSUE TITLE
Incorrect semver

### ISSUE THREAD

**Issue by jaydenseric** [zeit/next.js#5976]
# Bug report

## Describe the bug

Many recent prereleases, including the latest (v7.0.2-canary.50) have incorrect semver. Look at [the `next` releases on npm](https://www.npmjs.com/package/next) and immediately see that they are wrong. Versions are being published with semver that comes before already released versions.

## To Reproduce

To understand the consequences, create a package.json with this:

```json
{
  "dependencies": {
    "next": "^7.0.2-canary.50"
  }
}
```

Run `npm install && npm ls next` and you will see:

```
test@1.0.0 [redacted]
└── next@7.0.2
```

`v7.0.2` was installed, not `7.0.2-canary.50`!

That is because although v7.0.2 was published 3 months before v7.0.2-canary.50, according to semver it comes _after_, as stable releases always follow prereleases. Pre-releases are anything with a `-` after the `x.x.x`.

To make matters in this instance even more confusing, v7.0.3-alpha.5 was published before a whole bunch of v7.0.2-canary.x releases.

Once a stable patch has been published, move on to a prerelease for the next patch!

## Expected behavior

Releases should be correct semver.

## Additional context

This keeps happening again and again, despite multiple previous discussions acknowledging the issue: https://github.com/zeit/next.js/issues/4955. @timneutkens it is very frustrating lessons of the past are forgotten.

**Comment by Enalmada** [comment]
I had `"next": "^7.0.2-canary.xx"` in my package.json and thought I was on the latest version for quite a while before realizing I wasn't getting some fixes I needed.  A very easy mistake to make!

**Comment by timneutkens** [comment]
> To make matters in this instance even more confusing, v7.0.3-alpha.5 was published before a whole bunch of v7.0.2-canary.x releases.

This was a testing release and should be ignored in every sense.

**Comment by timneutkens** [comment]
The version scheme we had used `release`, we then switched to a monorepo using `lerna` to release prerelease versions. Then a security patch had to be released against old code / versioning scheme while the new scheme was being used. Hence why we haven't changed the versions yet. I'll release a new canary version today to solve this, this is a totally different issue than before btw, as that was bug/issue in `release` itself.
